src/counts.jl

```diff
@@ -167,38 +207,50 @@ proportions(x::IntegerArray, y::IntegerArray, wv::WeightVec) = proportions(x, y,
 ## auxiliary functions
 
 function _normalize_countmap{T}(cm::Dict{T}, s::Real)
-	r = Dict{T,Float64}()
-	for (k, c) in cm
-		r[k] = c / s
-	end
-	return r
+    r = Dict{T,Float64}()
+    for (k, c) in cm
+        r[k] = c / s
+    end
+    return r
 end
 
 ## 1D
 
 function addcounts!{T}(cm::Dict{T}, x::AbstractArray{T})
-	for v in x
-		cm[v] = get(cm, v, 0) + 1
-	end
-	return cm
+    for v in x
+        cm[v] = get(cm, v, 0) + 1
+    end
+    return cm
 end
 
 function addcounts!{T,W}(cm::Dict{T}, x::AbstractArray{T}, wv::WeightVec{W})
-	n = length(x)
-	length(wv) == n || raise_dimerror()
-	w = values(wv)
-	z = zero(W)
-
-	for i = 1 : n
-		@inbounds xi = x[i]
-		@inbounds wi = w[i]
-		cm[xi] = get(cm, xi, z) + wi
-	end
-	return cm
+    n = length(x)
+    length(wv) == n || throw(DimensionMismatch())
+    w = values(wv)
+    z = zero(W)
+
+    for i = 1 : n
+        @inbounds xi = x[i]
+        @inbounds wi = w[i]
+        cm[xi] = get(cm, xi, z) + wi
+    end
+    return cm
 end
 
+
+"""
+    countmap(x) -> Dict
+
+Return the counts of each distinct element of `x` as a dictionary.
```

Comment:
Could be more specific: "Return a dictionary mapping each unique value in `x` to its number of occurrences".